java/client/src/main/java/glide/api/models/BaseTransaction.java

```diff
@@ -2842,6 +2844,47 @@ public T geopos(@NonNull String key, @NonNull String[] members) {
         return getThis();
     }
 
+    /**
+     * Returns the distance between <code>member1</code> and <code>member2</code> saved in the
+     * geospatial index stored at <code>key</code>.
+     *
+     * @see <a href="https://valkey.io/commands/geodist">valkey.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param member1 The name of the first member.
+     * @param member2 The name of the second member.
+     * @param geoUnit The unit of distance measurement {@link GeoUnit}.
+     * @return Command Response - The distance between <code>member1</code> and <code>member2</code>.
+     *     If one or both members do not exist or if the key does not exist returns <code>null</code>.
+     */
+    public T geodist(
+            @NonNull String key,
+            @NonNull String member1,
+            @NonNull String member2,
+            @NonNull GeoUnit geoUnit) {
+        ArgsArray commandArgs =
+                buildArgs(concatenateArrays(new String[] {key, member1, member2, geoUnit.getRedisApi()}));
+        protobufTransaction.addCommands(buildCommand(GeoDist, commandArgs));
+        return getThis();
+    }
+
+    /**
+     * Returns the distance between <code>member1</code> and <code>member2</code> saved in the
+     * geospatial index stored at <code>key</code>.
+     *
+     * @see <a href="https://valkey.io/commands/geodist">valkey.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param member1 The name of the first member.
+     * @param member2 The name of the second member.
+     * @return Command Response - The distance between <code>member1</code> and <code>member2</code>.
+     *     If one or both members do not exist or if the key does not exist returns <code>null</code>.
+     *     The default unit is <code>METERS</code>.
+     */
+    public T geodist(@NonNull String key, @NonNull String member1, @NonNull String member2) {
+        ArgsArray commandArgs = buildArgs(new String[] {key, member1, member2});
```

Comment:
nit: you can omit array creation
```suggestion
        ArgsArray commandArgs = buildArgs(key, member1, member2);
```